tests/e2e/api_event_receiver_test.go

```diff
@@ -0,0 +1,109 @@
+package e2e
+
+import (
+	"encoding/json"
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/sassoftware/event-provenance-registry/tests/common"
+	"gotest.tools/v3/assert"
+)
+
+func TestCreateAndGetReceiver(t *testing.T) {
+	client := common.NewHTTPClient()
+
+	tests := map[string]struct {
+		input eventReceiverInput
+	}{
+		"empty schema": {
+			input: eventReceiverInput{
+				Name:        "has-no-schema",
+				Type:        "artifact.deploy.eks",
+				Version:     "0.0.5",
+				Description: "deploy to eks",
+				Schema:      `{}`,
+			},
+		},
+		"basic schema": {
+			input: eventReceiverInput{
+				Name:        "has-basic-schema",
+				Type:        "artifact.deploy.gcp",
+				Version:     "0.3.2",
+				Description: "deploy to gcp",
+				Schema:      `{"type":"object","properties":{"region":{"type":"string"}}}`,
+			},
+		},
+	}
+
+	for testName, tt := range tests {
+		t.Run(testName, func(t *testing.T) {
+			resp, err := client.Post(receiverURI, "application/json", strings.NewReader(tt.input.toPayload()))
+			assert.NilError(t, err)
+			assert.Equal(t, resp.StatusCode, http.StatusOK)
+
+			var postBody postReceiverResponse
+			err = json.NewDecoder(resp.Body).Decode(&postBody)
+			assert.NilError(t, err)
+			assert.Equal(t, len(postBody.Errors), 0, "got error(s) %v", postBody.Errors)
+			assert.Check(t, len(postBody.Data) > 0, "expect non-empty id")
+
+			receiverID := postBody.Data
+			resp, err = client.Get(receiverURI + receiverID)
+			assert.NilError(t, err)
+			assert.Equal(t, resp.StatusCode, http.StatusOK)
+
+			var getBody getReceiverResponse
+			err = json.NewDecoder(resp.Body).Decode(&getBody)
+			assert.NilError(t, err)
+			assert.Equal(t, len(getBody.Errors), 0)
+			assert.Equal(t, len(getBody.Data), 1)
+			receiver := getBody.Data[0]
+			assert.Equal(t, string(receiver.ID), receiverID)
+			assert.Equal(t, receiver.Name, tt.input.Name)
+			assert.Equal(t, receiver.Type, tt.input.Type)
+			assert.Equal(t, receiver.Version, tt.input.Version)
+			assert.Equal(t, receiver.Description, tt.input.Description)
+			assert.Equal(t, receiver.Schema.String(), tt.input.Schema)
+			assert.Check(t, len(receiver.Fingerprint) > 0, "expect non-empty fingerprint")
+			assert.Check(t, !time.Time(receiver.CreatedAt.Date).IsZero(), "expect time to be set")
+		})
+	}
+}
+
+func TestCreateInvalidReceiver(t *testing.T) {
+	client := common.NewHTTPClient()
+
+	receiver := eventReceiverInput{
+		Name:        "upload artifact",
```

Comment:
Add some cases with those fields empty. I believe they're all required.

Reply:
Are we sure all fields for events, receivers, & groups are required? I wanted to save those test cases for when we had a better idea of what's valid. If you don't mind, I'd prefer those cases be added in #70.

Reply:
Not every field is required, but I believe the ones you have defined are. I'm fine leaving it for 70.